### PR TITLE
OSDOCS-11236: Add guidance about dual-stack networking limitations

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.16 + 

Issue:
[OSDOCS-11236](https://issues.redhat.com/browse/OSDOCS-11236)

Link to docs preview:
[Network configuration parameters](https://79422--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installation-config-parameters-bare-metal#installation-configuration-parameters-network_installation-config-parameters-bare-metal)

- [x] SME has approved this change.
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -192,7 +192,12 @@ Only IPv4 addresses are supported.
 endif::agent,bare,ibm-power,ibm-z,vsphere[]
 
 ifdef::agent,bare,ibm-power,ibm-z,vsphere[]
+Consider the following information before you configure network parameters for your cluster:
```
Comment (dfitzmau) [inline modules/installation-configuration-parameters.adoc:195]: According to [Glenn's comment](https://issues.redhat.com/browse/OCPBUGS-34504#:~:text=Two%20solutions%20for%20this%3A) on this Jira, should I mention DHCP?

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -192,7 +192,12 @@ Only IPv4 addresses are supported.
...
+
 * If you use the {openshift-networking} OVN-Kubernetes network plugin, both IPv4 and IPv6 address families are supported.
+** If you want a host with a non-IPv6 network configuration to access a cluster with a configured IPv6 network, you can configure the cluster's firewall to block traffic for the IPv6 protocol.
+* If you deployed nodes in an {product-title} cluster with a network that supports both IPv4 and non-link-local IPv6 addresses, with the exception of `fe80::`, configure your cluster to use a dual-stack network.
```
Comment (dfitzmau) [inline modules/installation-configuration-parameters.adoc:199]: The RFC documentation is confusing in that fe80: is jumbled in with link-local addresses:

Screenshot from 2024-07-25 12-07-59

https://datatracker.ietf.org/doc/rfc7404/

Stating `fe80::` as a non-LLA is correct?

File: modules/installation-configuration-parameters.adoc (hunk 1 above)
Comment (cybertron) [reply modules/installation-configuration-parameters.adoc:195]: We could mention that if you have v6 DHCP in your infrastructure and don't want to use v6 in your cluster you can disable v6 on the primary interface using NMState.

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -192,7 +192,12 @@ Only IPv4 addresses are supported.
 endif::agent,bare,ibm-power,ibm-z,vsphere[]
 
 ifdef::agent,bare,ibm-power,ibm-z,vsphere[]
+Consider the following information before you configure network parameters for your cluster:
+
 * If you use the {openshift-networking} OVN-Kubernetes network plugin, both IPv4 and IPv6 address families are supported.
+** If you want a host with a non-IPv6 network configuration to access a cluster with a configured IPv6 network, you can configure the cluster's firewall to block traffic for the IPv6 protocol.
```
Comment (cybertron) [inline modules/installation-configuration-parameters.adoc:198]: Is there more context for this? I'm a little confused why this would be necessary. If you have a v4 client talking to a dual stack cluster, it's always going to talk to the v4 endpoints because that's all it has access to.

If you don't want to allow v6 access to your cluster at all, you do have the option of not configuring v6 VIPs so the only way for external clients to talk to it is v4, but I'm not sure we need to call that out explicitly.

File: modules/installation-configuration-parameters.adoc (hunk 2 above)
Comment (cybertron) [reply modules/installation-configuration-parameters.adoc:199]: fe80:: is link-local: https://en.wikipedia.org/wiki/Link-local_address#IPv6

I think for the purposes of this doc, just saying non-link-local is sufficient. That should make it clear that the link-local address you get in ipv4 deployments is fine, but that if you're doing something more than that you need to use dual stack.

File: modules/installation-configuration-parameters.adoc (hunk 1 above)
Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc:195]: Thanks. Added:

> If your infrastructure uses IPv6 as the primary network layer protocol in a Dynamic Host Configuration Protocol (DHCP) environment and your cluster does not support IPv6, you can use the Kubernetes NMState Operator to disable IPv6 on your infrastructure

File: modules/installation-configuration-parameters.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc:200]: :heavy_check_mark:

File: modules/installation-configuration-parameters.adoc (hunk 3 above)
Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc]: Removed the line. 

> If you want a host with a non-IPv6 network configuration to access a cluster with a configured IPv6 network, you can configure the cluster's firewall to block traffic for the IPv6 protocol.

File: modules/installation-configuration-parameters.adoc [hunk 4]
```diff
@@ -192,7 +192,12 @@ Only IPv4 addresses are supported.
 endif::agent,bare,ibm-power,ibm-z,vsphere[]
 
 ifdef::agent,bare,ibm-power,ibm-z,vsphere[]
+Consider the following information before you configure network parameters for your cluster:
+
 * If you use the {openshift-networking} OVN-Kubernetes network plugin, both IPv4 and IPv6 address families are supported.
+* If your infrastructure uses IPv6 as the primary network layer protocol in a Dynamic Host Configuration Protocol (DHCP) server and your cluster does not support IPv6, you can use the Kubernetes NMState Operator to disable IPv6 on your infrastructure.
```
Comment (cybertron) [inline modules/installation-configuration-parameters.adoc:198]: I'm not sure you can use the operator for this, I think it needs to be a day 1 config. Otherwise the nodes will initially come up with ipv6 enabled and may cause problems.

Given that we're starting to support dual stack on non-baremetal platforms, maybe we should not specify a method at all since it may vary depending on platform. We could shorten this to: "If your infrastructure uses IPv6 in a Dynamic Host Configuration Protocol (DHCP) server and your cluster does not support IPv6, you can disable IPv6 on your nodes' interfaces."

Oh, and as I'm reading this closer, it doesn't matter if v6 is primary or not so I think we should drop that as well. The updated line above includes all of the changes I would suggest here.

Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc]: Updated. 

I'm not sure if "Oh, and as I'm reading this closer, it doesn't matter if v6 is primary or not so I think we should drop that as well. The updated line above includes all of the changes I would suggest here." indicates I should remove one of the bullet points or an IPv6 reference?

**Review by cybertron — COMMENTED** [review]
/lgtm

**Review by xenolinux — COMMENTED** [review]
Nice! 
LGTM

/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -192,7 +192,12 @@ Only IPv4 addresses are supported.
 endif::agent,bare,ibm-power,ibm-z,vsphere[]
 
 ifdef::agent,bare,ibm-power,ibm-z,vsphere[]
+Consider the following information before you configure network parameters for your cluster:
+
 * If you use the {openshift-networking} OVN-Kubernetes network plugin, both IPv4 and IPv6 address families are supported.
+* If your infrastructure uses IPv6 in a Dynamic Host Configuration Protocol (DHCP) server and your cluster does not support IPv6, you can disable IPv6 on your infrastructure.
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc:198]: I checked with Ben, I'll remove:

> If your infrastructure uses IPv6 in a Dynamic Host Configuration Protocol (DHCP) server and your cluster does not support IPv6, you can disable IPv6 on your infrastructure.

[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/installation-configuration-parameters.adoc:198]: The potential to do this configuration is a customer's choice and not something we should document.

**Review by cybertron — COMMENTED** [review]
/lgtm

**Review by jeana-redhat — APPROVED** [review]
LGTM

/remove-label merge-review-in-progress
/remove-label merge-review-needed
